### PR TITLE
[icinga2_host.py] Actually return codes instead of data

### PR DESCRIPTION
**Pull request by C0rn3j**
##### SUMMARY
Currently the module tries to return the `data`, which can result in a blank message instead of the code being shown.

```
 "msg": "bad return code creating host: "
```

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
icinga2_host.py

##### ADDITIONAL INFORMATION
Trying to access an API with invalid cert and `validate_certs` set to `no` will reproduce this.

### CODE REVIEW

**Comment by felixfontein** [comment]
You need a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to).

**Comment by C0rn3j** [comment]
Added

File: changelogs/fragments/335-icinga2_host-return-error-code.yaml
Comment (Andersson007) [inline changelogs/fragments/335-icinga2_host-return-error-code.yaml:2]:
```suggestion
  - icinga2_host - fix returning error codes (https://github.com/ansible-collections/community.general/pull/335).
```

**Comment by Andersson007** [comment]
@t794104 could you please take a look?

**Comment by gundalow** [comment]
@t794104 Would you be able to take a look at this bug fix for `icinga2_host` Ansible module?

**Comment by dericcrago** [comment]
Tested locally:

```
# before
"msg": "bad return code creating host: "
```

```
# after
"msg": "bad return code creating host: -1"
```

File: plugins/modules/monitoring/icinga2_host.py
Comment (felixfontein) [inline plugins/modules/monitoring/icinga2_host.py:285]:
```suggestion
                        module.fail_json(msg="bad return code deleting host: %s %s" % (ret['code'], ret['data']))
```

File: plugins/modules/monitoring/icinga2_host.py
Comment (felixfontein) [inline plugins/modules/monitoring/icinga2_host.py:301]:
```suggestion
                module.fail_json(msg="bad return code modifying host: %s %s" % (ret['code'], ret['data']))
```

File: plugins/modules/monitoring/icinga2_host.py
Comment (felixfontein) [inline plugins/modules/monitoring/icinga2_host.py:313]:
```suggestion
                        module.fail_json(msg="bad return code creating host: %s %s" % (ret['code'], ret['data']))
```

File: plugins/modules/monitoring/icinga2_host.py
Comment (dericcrago) [reply plugins/modules/monitoring/icinga2_host.py:285]: @felixfontein how bout something like this to maybe clarify which code is which if you only get one?

```suggestion
                        module.fail_json(msg="bad return code (%s) deleting host: '%s'" % (ret['code'], ret['data']))
```

Comment (felixfontein) [reply plugins/modules/monitoring/icinga2_host.py:285]: @dericcrago that would be even better!

Comment (felixfontein) [reply plugins/modules/monitoring/icinga2_host.py:285]: I adjusted the other two suggestions approriately, feel free to commit yours and them if you think it's good.

**Comment by dericcrago** [comment]
tested locally:

```
# after
"msg": "bad return code (-1) creating host: ''"
```

**Review by felixfontein — APPROVED** [review]

**Review by Andersson007 — APPROVED** [review]

**Comment by Andersson007** [comment]
@C0rn3j after this is merged, would you like to be added as a maintainer of the module to `.github/BOTMETA.yml`?
You will get notifications about issues / PRs related to the module (i don't think there will be many) and your `shipit`/`LGTM` will be counted by the bot for authomerge of fixes / new features (needs two). What do you think?

**Comment by Andersson007** [comment]
Merged
@C0rn3j thanks for the fix (and see my comment above). @felixfontein @gundalow @dericcrago thanks for reviewing!

**Pull request merged**
